src/ui/MLFacet/MLFacetHeader/MLFacetHeader.ts

```diff
@@ -38,7 +39,13 @@ export class MLFacetHeader {
 
   private clear() {
     this.facet.reset();
-    this.facet.triggerNewQuery();
+    this.facet.triggerNewQuery(() =>
+      this.facet.sendUsageAnalyticsEvent(analyticsActionCauseList.mLFacetClearAll, {
```

Comment:
Why pass an argument `beforeExecuteQuery` to `triggerNewQuery` rather than logging inside the `clear` method and then triggering a new query? e.g.

```
private clear() {
    this.facet.reset();
    this.logClearAllToAnalytics();
    this.facet.triggerNewQuery();
}
```

It would also be good to extract the function out into a method.